apollo-coroutines-support/src/main/kotlin/com/apollographql/apollo/coroutines/CoroutinesExtensions.kt

```diff
@@ -141,35 +139,34 @@ fun <T> ApolloSubscriptionCall<T>.toFlow(): Flow<Response<T>> = callbackFlow {
 }
 
 /**
- * Converts an [ApolloPrefetch] to [Job].
+ * Suspends the [ApolloPrefetch] until it completes and returns the value on success or throws an exception on failure.
+ * The [ApolloPrefetch] is cancelled when the coroutine running the operation is cancelled as well.
  *
  * @param <T>  the value type.
- * @return the converted job
+ * @return the response on success.
+ * @throws ApolloException on failure.
  */
-fun ApolloPrefetch.toJob(): Job {
-  val deferred = CompletableDeferred<Unit>()
-
-  deferred.invokeOnCompletion {
-    if (deferred.isCancelled) {
-      cancel()
-    }
-  }
+suspend fun ApolloPrefetch.await(): Unit = suspendCancellableCoroutine { cont ->
 
   enqueue(object : ApolloPrefetch.Callback() {
+
+    private var resumedIsCalled = false
```

Comment:
I'm not familiar with this library tbh, but reading the documentation for `ApolloPrefetch` says it should only be called back once so you shouldn't need this. Then again adding the check doesn't hurt. I'll let library maintainers say whether this can be called multiple times or not.

Reply:
I'm not 100% sure either. I'd vote to play it safe and keep the guard there.